### PR TITLE
INT-4100: Improve `RequestReplyExchanger`

### PR DESCRIPTION
**Pull request by artembilan**
JIRA: https://jira.spring.io/browse/INT-4100

* Add `throws MessagingException` to the `RequestReplyExchanger` to make it fully messaging contract based
* Add test to the `ServiceActivatorDefaultFrameworkMethodTests` to demonstrate how many info is in the stack trace after `RequestReplyExchanger`
* Document how to be with the `throws` clause in custom gateways
* Fix `boolean` attributes in the `spring-integration-5.0.xsd`

### CODE REVIEW

File: spring-integration-core/src/test/java/org/springframework/integration/handler/ServiceActivatorDefaultFrameworkMethodTests-context.xml [hunk 1]
```diff
@@ -42,9 +42,9 @@
 
 	<gateway id="gateway" default-request-channel="requestChannel" default-reply-channel="replyChannel"/>
 
-	<channel id="requestChannel"/>
-
-	<bridge id="bridge" input-channel="requestChannel" output-channel="replyChannel"/>
+	<chain id="bridge" input-channel="requestChannel" output-channel="replyChannel">
```
Comment (garyrussell) [inline spring-integration-core/src/test/java/org/springframework/integration/handler/ServiceActivatorDefaultFrameworkMethodTests-context.xml]: We should change the id - it's no longer a bridge.

**Review by garyrussell — COMMENTED** [review]
That's all, but we need a what's new for the `RequestReplyExchanger` behavior change (for people who are using it and expect the exception to be unwrapped) - advise them to use their own interface. 

Also a migration guide entry.

**Comment by garyrussell** [comment]
Hmmm - you didn't address... `... but we need a what's new ...`

**Comment by artembilan** [comment]
Also see https://github.com/spring-projects/spring-integration/wiki/Spring-Integration-4.3-to-5.0-Migration-Guide#requestreplyexchanger-contract

**Comment by garyrussell** [comment]
Doc polishing here for your review https://github.com/garyrussell/spring-integration/commit/a7feb979ab46311ddd712756a5f18b2fa944c5a8

**Comment by garyrussell** [comment]
Merged as 8da89c7cce0cdd20407f5a9ae352d0c9923c16dc

**Pull request closed without merging**